src/Public/OpenApi.ps1

```diff
@@ -495,871 +772,2452 @@ Creates a Request Body definition for routes.
             }
+
+            $podeRoot = Get-PodeModuleMiscPath
+            Write-PodeFileResponse -Path ([System.IO.Path]::Combine($podeRoot, 'default-swagger-editor.html.pode')) -Data $Data
+        }
+
+        $swaggerEditorPath = Join-Path -Path $(Get-PodeModuleMiscPath) -ChildPath 'swagger-editor-dist'
+        Add-PodeStaticRoute -Path  $meta.SwaggerEditorDist -Source $swaggerEditorPath -EndpointName $EndpointName
+
+        $PodeContext.Server.OpenAPI.Definitions[$DefinitionTag].hiddenComponents.viewer['editor'] = $Path
+    } elseif ($Bookmarks.IsPresent) {
+        # set a default path
+        $Path = Protect-PodeValue -Value $Path -Default '/docs'
+        if ([string]::IsNullOrWhiteSpace($Title)) {
+            throw "No route path supplied for $($Type) page"
+        }
+        # setup meta info
+        $meta = @{
+            Title         = $Title
+            OpenApi       = $OpenApiUrl
+            DarkMode      = $DarkMode
+            DefinitionTag = $DefinitionTag
         }
 
-        'reference' {
-            if (!(Test-PodeOAComponentRequestBody -Name $Reference)) {
-                throw "The OpenApi component request body doesn't exist: $($Reference)"
+        $route = Add-PodeRoute -Method Get -Path $Path `
+            -Middleware $Middleware -ArgumentList $meta `
+            -EndpointName $EndpointName -Authentication $Authentication `
+            -Role $Role -Scope $Scope -Group $Group `
+            -PassThru -ScriptBlock {
+            param($meta)
+            $Data = @{
+                Title   = $meta.Title
+                OpenApi = $meta.OpenApi
+            }
+            $DefinitionTag = $meta.DefinitionTag
+            foreach ($type in $PodeContext.Server.OpenAPI.Definitions[$DefinitionTag].hiddenComponents.viewer.Keys) {
+                $Data[$type] = $true
+                $Data["$($type)_path"] = $PodeContext.Server.OpenAPI.Definitions[$DefinitionTag].hiddenComponents.viewer[$type]
+            }
+
+            $podeRoot = Get-PodeModuleMiscPath
+            Write-PodeFileResponse -Path ([System.IO.Path]::Combine($podeRoot, 'default-doc-bookmarks.html.pode')) -Data $Data
+        }
+        if (! $NoAdvertise.IsPresent) {
+            $PodeContext.Server.OpenAPI.Definitions[$DefinitionTag].hiddenComponents.bookmarks = @{
+                path       = $Path
+                route      = @()
+                openApiUrl = $OpenApiUrl
             }
+            $PodeContext.Server.OpenAPI.Definitions[$DefinitionTag].hiddenComponents.bookmarks.route += $route
+        }
 
-            return @{
-                '$ref' = "#/components/requestBodies/$($Reference)"
+    } else {
+        if ($Type -ieq 'RapiPdf' -and (Test-PodeOAVersion -Version 3.1 -DefinitionTag $DefinitionTag)) {
+            throw "The Document tool RapidPdf doesn't support OpenAPI 3.1"
+        }
+        # set a default path
+        $Path = Protect-PodeValue -Value $Path -Default "/docs/$($Type.ToLowerInvariant())"
```

Comment:
Doing some testing, and just spotted that this line has a breaking change in it. the default path should be the following, without the `/docs`:

```powershell
-Default "/$($Type.ToLowerInvariant())"
```

It's probably worth updating the Bookmarks and Editor to follow the same pattern. Ie:

```powershell
-Default '/bookmarks'
-Default '/editor'
```